openproblems/tasks/denoising/methods/magic.py

```diff
@@ -14,18 +14,35 @@
     code_url="https://github.com/KrishnaswamyLab/MAGIC",
     image="openproblems-python-extras",
 )
+_knn_naive_method = functools.partial(
+    method,
+    paper_name="NA",
+    paper_url="NA",
+    paper_year=2022,
+    code_url="https://openproblems.bio",
+    image="openproblems-python-extras",
+)
 
 
-def _magic(adata, solver):
+def _magic(adata, solver, normtype, decay, t):
     from magic import MAGIC
 
     X, libsize = scprep.normalize.library_size_normalize(
         adata.obsm["train"], rescale=1, return_library_size=True
     )
-    X = scprep.transform.sqrt(X)
-    Y = MAGIC(solver=solver, verbose=False).fit_transform(X, genes="all_genes")
-    Y = scprep.utils.matrix_transform(Y, np.square)
-    Y = scprep.utils.matrix_vector_elementwise_multiply(Y, libsize, axis=0)
+
+    if (normtype == "sqrt"):
+        X = scprep.transform.sqrt(X)
```

Comment:
Instead of actually running the norm here, I'd recommend defining
```
if normtype == "sqrt":
    norm_fn = np.sqrt
    denorm_df = np.square
elif normtype == "log":
    norm_fn = np.log1p
    denorm_df = np.expm1

X = scprep.utils.matrix_transform(X, norm_fn)
...
Y = scprep.utils.matrix_transform(Y, denorm_fn)
```